src/test/java/com/squareup/phrase/PhraseTest.java

```diff
@@ -116,6 +116,14 @@
         .isEqualTo("Hello Eric");
   }
 
+  @Test public void putFailUppercaseNotAllowed() {
+    try{
+      from("Hello {Name}").putOptional("Name", "Eric").format().toString();
+      fail("Expected IllegalArgumentException");
+    } catch (IllegalArgumentException e) {
+    }
```

Comment:
I think the key here is to verify that the value of `e.getMessage()` is not `Missing Closing Brace`